pywr-core/src/agg_funcs.rs

```diff
@@ -0,0 +1,389 @@
+use crate::recorders::PeriodValue;
+
+/// Aggregation functions that can be applied to a set of f64 values.
+#[derive(Clone, Debug)]
+pub enum AggFuncF64 {
+    Sum,
+    Mean,
+    Product,
+    Min,
+    Max,
+    CountNonZero,
+    CountFunc { func: fn(f64) -> bool },
+}
+
+impl AggFuncF64 {
+    /// Calculate the aggregation of the given values.
+    pub fn calc_period_values(&self, values: &[PeriodValue<f64>]) -> Option<f64> {
+        match self {
+            Self::Sum => Some(values.iter().map(|v| v.value * v.duration.fractional_days()).sum()),
```

Comment:
Is there a reason why fractional days is not accounted for in the Product, Min and Max variants? I might be missing something here...